presto-spi/src/main/java/com/facebook/presto/spi/SplitContext.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class SplitContext
+{
+    public static final SplitContext NON_CACHEABLE = new SplitContext(false);
+
+    private final boolean cacheable;
+
+    @JsonCreator
+    public SplitContext(@JsonProperty boolean cacheable)
```

Comment:
nit: maybe we'd want a builder pattern for this class? Even though currently there is only one field, a builder pattern makes future work easier to understand.

Reply:
Currently I couldn't come up with anything that we may need to add to SplitContext, maybe we can use it when we have additional optional attribute for SplitContext
thanks for the good advice, left a todo for future change